ios/RNSScreen.mm

```diff
@@ -246,6 +246,17 @@ - (void)setStackPresentation:(RNSScreenStackPresentation)stackPresentation
       }
 #else
       _controller.modalPresentationStyle = UIModalPresentationFullScreen;
+#endif
+#if !TARGET_OS_TV
+      if (@available(iOS 18.0, *)) {
+        UISheetPresentationController *sheetController = _controller.sheetPresentationController;
+        if (sheetController != nil) {
+          sheetController.prefersPageSizing = true;
+        } else {
+          RCTLogError(
+              @"[RNScreens] sheetPresentationController is null when attempting to set prefersPageSizing for modal");
+        }
+      }
```

Comment:
This check for iOS 18 won't be enough. If we leave this in current shape it'll fail to compile when somebody uses XCode / SDK that does not support iOS 17 or greater. Sometimes people are limited due to hardware limitations or business reasons.

What we should do here is to wrap it in

```
#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_17_0) && \
    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_17_0
```